07_safety/juvenile-arrests/juvenile-arrests-place-all.Rmd

```diff
@@ -0,0 +1,876 @@
+  )
+
+## Recode ethnicity
+arrests <- arrests %>%
+  mutate(
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="h", "hispanic origin", ethnicity_of_arrestee),
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="n", "not of hispanic origin", ethnicity_of_arrestee),
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="u", "unknown", ethnicity_of_arrestee)
+  )
+
+```
+
+58,052 observations have `unknown` ethnicity and 75,253 have missing ethnicity
+```{r}
+arrests %>% 
+  group_by(race_of_arrestee, ethnicity_of_arrestee) %>% 
+  count()
+
+arrests %>% 
+  group_by(ethnicity_of_arrestee) %>% 
+  count()
+
+```
+
+
+### 3.4 Aggregate arrest counts to the reporting agency level
+Aggregate arrests by agency and finish race construction
+
+Note that this metric includes the following subgroups by race, sex, and age subgroups, so those subgroups are also aggregated in this section:
+    * Race: white, Black, Hispanic, Asian/other
+    * Sex: male, female
+    * Age: ages 10-14, ages 15-17
+```{r}
+arrests_agency <- arrests %>%
+  filter(age >= 10) %>%
+  group_by(year, ori) %>%
+  summarize(
+    arr_total = n(),
+    # arr_prop = sum(property==1),
+    # arr_viol = sum(violent==1),
+    # Race subgroups
+    race_aian = sum(race_of_arrestee == "american indian/alaskan native"),
+    race_asian = sum(race_of_arrestee == "asian"),
+    race_black = sum(race_of_arrestee == "black"),
+    race_nhpi = sum(race_of_arrestee == "native hawaiian or other pacific islander"),
+    race_unkno = sum(race_of_arrestee == "unknown"),
+    race_white = sum(race_of_arrestee == "white"),
+    ethn_hispanic = sum(ethnicity_of_arrestee == "hispanic origin"),
+    ethn_nonhisp = sum(ethnicity_of_arrestee == "not of hispanic origin"),
+    # Sex subgroups
+    sex_male = sum(sex_of_arrestee == "male"),
+    sex_female = sum(sex_of_arrestee == "female"),
+    # Age subgroups
+    age_10_14 = sum(age >= 10 & age <= 14),
+    age_15_17 = sum(age >= 15 & age <= 17),
+    # Aggregating group as well
+    group_a = max(group=="A"),
+    group_b = max(group=="B")
+  ) %>%
+  ungroup() %>%
+  # Construct last race/ethnicity subgroup
+  mutate(race_asian_other = race_aian + race_asian + race_nhpi)
+
+```
+
+
+## 4. Aggregate agency-level arrests to the place level
+This section aggregates the agency-level arrest counts from the agency-level to the place-level.
+
+Our data are currently long by year/agency. However, we want to know the arrest counts at the place level, not the agency level. In the `01_agency_geo_place.Rmd` program, we constructed a file that linked reporting agencies to places. We can use that file to aggregate counts from agencies to places.
+
+### 4.1 Load universe of place/agency pairs
+First load the universe of agencies linked to places. This file was created in `01_agency_geo_place.Rmd` and is long by unique agency/place pairs.
+```{r}
+bhf_place_urb <- read_csv(here::here("07_safety", "modified data", "all_agency_place.csv"))
```

Comment:
similar to the violent crime file, I think you need to add a year variable to account for the agencies that don't match 
`bhf_place_urb <- read_csv(here::here("07_safety", "modified data", "all_agency_place.csv")) %>% 
  expand_grid(year = c(2021, 2022))`

Reply:
I edited this in the `01_agency_geo_place.qmd` file